### PR TITLE
replace validate_* calls with datatypes

### PR DESCRIPTION
**Pull request by bastelfreak**

### CODE REVIEW

File: manifests/server.pp [hunk 1]
```diff
@@ -391,183 +391,123 @@
...
-
-  validate_re($implementation, '^(master|puppetserver)$')
-  validate_re($parser, '^(current|future)$')
-
   if $environment_timeout {
     validate_re($environment_timeout, '^(unlimited|0|[0-9]+[smh]{1})$')
   }
 
   if $implementation == 'puppetserver' {
     validate_re($jvm_min_heap_size, '^[0-9]+[kKmMgG]$')
```
Comment (ekohl) [inline manifests/server.pp]: Given there are sane defaults, should this be enforced in the parameters as well?

Comment (bastelfreak) [inline manifests/server.pp]: Should be possible. I need to take a look at the datatypes documentation. I'm not sure how I can implement this.

Comment (ekohl) [inline manifests/server.pp]: https://docs.puppet.com/puppet/latest/lang_data_regexp.html#the-regexp-data-type tells me there should be a ```Regexp``` type for this.

Comment (bastelfreak) [inline manifests/server.pp]: I think this is correct:
https://docs.puppet.com/puppet/4.6/lang_data_abstract.html#pattern

**Comment by domcleal** [comment]
Replaced by #498, converting all parameters. Thanks @bastelfreak.

**Pull request closed without merging**